odoo/modules/registry.py

```diff
@@ -472,23 +475,33 @@ def check_indexes(self, cr, model_names):
                    [tuple(row[0] for row in expected)])
         existing = {row[0] for row in cr.fetchall()}
 
-        if not self.has_trigram and any(row[3] == 'trigram' for row in expected):
-            self.has_trigram = sql.install_pg_trgm(cr)
-
-        for indexname, tablename, columnname, index in expected:
+        for indexname, tablename, column_name, index, unaccent in expected:
             assert index in ('btree', 'btree_not_null', 'trigram', True, False, None)
             if index and indexname not in existing:
+                column_expression = f'"{column_name}"'
```

Comment:
Use [`psycopg2.extensions.quote_ident`](https://www.psycopg.org/docs/extensions.html#psycopg2.extensions.quote_ident).

Reply:
Let's do that everywhere, but in a specific pull request.  That's not the point of this patch.

Reply:
This specific line is getting extracted / added, I don't see the issue of fixing it while at it.

Reply:
You mean `sql.Identifier(column_name)` ?

Reply:
> You mean `sql.Identifier(column_name)` ? If it is the case, yes why not.

No I mean `quote_ident`, switching to `sql.Identifier` would also require switching `create_index` and all its caller to using the `psycopg2.sql` helpers which is a lot more work.

Reply:
Ok I see.
IMO, `quote_ident` is very ugly in Odoo : `quote_ident(column_name, cr._obj)` (the `_obj` or `_cnx`...)

Reply:
> Ok I see. IMO, `quote_ident` is very ugly in Odoo : `quote_ident(column_name, cr._obj)` (the `_obj` or `_cnx`...)

If that's a concern we could just add a helper for that on `Cursor`.

Reply:
Yep of course. It can clearly be a clean up to do.